src/App.tsx

```diff
@@ -1,24 +1,208 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoAppHeader } from './components/TodoAppHeader/TodoAppHeader';
+import { TodoAppContent } from './components/TodoAppContent/TodoAppContent';
+import { TodoAppFooter } from './components/TodoAppFooter/TodoAppFooter';
+import { Notifications } from './components/Notifications/Notifications';
+import { client } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
+import { TodoPostData } from './types/TodoPostData';
 
-const USER_ID = 0;
+const USER_ID = 10308;
+
+const getTodos = () => client.get<Todo[]>(`/todos?userId=${USER_ID}`);
+const postTodo = (data: TodoPostData) => client.post<Todo>(`/todos?userId=${USER_ID}`, data);
+const deleteTodo = (todoId: number) => client.delete(`/todos/${todoId}`);
+
+const prepareTodos = (todoList: Todo[], filterType: FilterType) => (
+  todoList.filter(todo => {
+    switch (filterType) {
+      case FilterType.ACTIVE:
+        return !todo.completed;
+
+      case FilterType.COMPLETED:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  })
+);
+
+const getActiveTodosCount = (todoList: Todo[]) => (
+  todoList.filter(todo => !todo.completed).length
+);
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[] | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.ALL);
+  const [errorType, setErrorType] = useState(ErrorType.NONE);
+  const [isErrorShown, setIsErrorShown] = useState(false);
+  const [todoInputValue, setTodoInputValue] = useState('');
```

Comment:
you can move it to the item

Reply:
Sorry, didn't actually understand which from these can be moved to item. All of these hooks are used in App component logic.